### PR TITLE
Retry to create storage container.

### PR DESCRIPTION
**Pull request by genevieve**
https://github.com/terraform-providers/terraform-provider-azurerm/issues/841

### CODE REVIEW

File: azurerm/resource_arm_storage_container.go [hunk 1]
```diff
@@ -124,6 +125,18 @@ func resourceArmStorageContainerCreate(d *schema.ResourceData, meta interface{})
 	return resourceArmStorageContainerRead(d, meta)
 }
 
+func checkContainerIsCreated(reference *storage.Container) func() *resource.RetryError {
+	return func() *resource.RetryError {
+		createOptions := &storage.CreateContainerOptions{}
```
Comment (paultyng) [inline azurerm/resource_arm_storage_container.go]: Should we explicitly set timeout here so we can effectively use the timeout duration of the `Retry` call? I'm not sure on the default, maybe its ok?

Comment (genevieve) [reply azurerm/resource_arm_storage_container.go]: There’s sin inyl one other instance of retrying in the codebase that’s I found and the timeout was not configurable. We can add a create timeout to the scheme if we think it should be adjustable?

Comment (tombuildsstuff) [reply azurerm/resource_arm_storage_container.go]: we lean on the SDK's default timeout through the codebase, so I don't think we need to override this value? (2m should be enough time for Azure to become consistent)

**Comment by tombuildsstuff** [comment]
Tests pass:

screen shot 2018-02-16 at 09 45 39

**Comment by vchauhan1** [comment]
Still it's failing with the following error. azurerm_storage_container.be-sc: Error creating container "XXXXXXX" in storage account "XXXXXXXX": storage: service returned error: StatusCode=403, ErrorCode=AuthenticationFailed, ErrorMessage=Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.

**Comment by dariustehrani** [comment]
I can confirm this is still failing.

**Pull request merged**